### PR TITLE
AB: Microscopy Analyze Page

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?

Related to #202
#### What's this PR do?

Add microscopy analyze page to the Assignment Builder. The page lists all possible combinations of strain protocols and analysis type, grouped by analysis, condition combination.

### CODE REVIEW

**Comment by pdpinch** [comment]
@annagav I know this is still in progress, but I'd like to know which issues this addresses

**Comment by noisecapella** [comment]
After you merge #649 can you rebase this? I think I'm hitting the same bug you fixed there

**Comment by annagav** [comment]
@noisecapella, should be fixed now...

**Comment by noisecapella** [comment]
The back button on Microscopy - Analyze is going to Flow Cytometry - Sample Prep instead of Microscopy - Sample Prep

File: instructor/common.py [hunk 1]
```diff
@@ -1309,44 +1312,85 @@ def microscopy_sample_prep(request):
...
+            models.MicroscopyImageMapping.objects.get_or_create(
+                sample_prep=sample,
+                strain_protocol=strain_protocol,
+            )
+        )
+
 
+@assignment_selected
+@check_assignment_owner
+@login_required
```
Comment (noisecapella) [inline instructor/common.py:1333]: I missed this in the previous review, should these three decorators also go on `microscopy_sample_prep`?

File: instructor/common.py [hunk 2]
```diff
@@ -1309,44 +1312,85 @@ def microscopy_sample_prep(request):
...
+            if instance.instance.sample_prep.analysis == sample.analysis and
+            instance.instance.sample_prep.condition == sample.condition
+        ]
+
+        grouped_images.append(
+            (
+                sample.analysis, sample.condition, filtered_list
+            )
+        )
+    print grouped_images
```
Comment (noisecapella) [inline instructor/common.py]: Can you remove this line?

File: instructor/common.py (hunk 1 above)
Comment (annagav) [inline instructor/common.py:1333]: Yes, they should be there.

File: instructor/templates/instructor/micro_analyze.html [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                    <div class="scb_ab_s_row" style="padding: 10px 0 8px 0">
+
+                        <div class="scb_ab_s_col_width_number"></div>
+                        <div class="scb_ab_s_col_width_facs_sample">
+
+                            <img src="{{ instance.image.file.url }}" />
+                            <button type="button" class="scb_s_gray_button open_upload_window_btn"
+                                    data-row_id="row-{{ instance.id }}"
+                                    data-sample_treatment="{{ analysis }}, {{ condition }}">
+                                ADD IMAGE
```
Comment (noisecapella) [inline instructor/templates/instructor/micro_analyze.html:69]: Does this button do anything yet?

Comment (annagav) [inline instructor/templates/instructor/micro_analyze.html:69]: No, I am working on the select image window on different branch.

**Comment by annagav** [comment]
@noisecapella, I addressed the comments, and also I changed the view not to use formsets.

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**